lib/pymedphys/_ai/sql_agent/conversation.py

```diff
@@ -335,100 +159,65 @@ async def mosaiq_sql_agent(sub_agent_prompt: str):
     return tools_mapping
 
 
-async def _conversation_with_task_creation(
-    nursery: trio.Nursery,
-    tasks_record: list[TaskRecordItem],
+async def _conversation_with_tool_use(
     anthropic_client: AsyncAnthropic,
     connection: pymedphys.mosaiq.Connection,
     model: str,
     system_prompt: str,
-    message_send_channel: trio.MemorySendChannel[MessageParam],
-    messages: Messages,
+    tools: list[MessageParam],
+    messages: list[Message],
 ):
+    """Mutates messages in-place recursively.
+    NOTE: This is probably not a good idea, likely worth refactoring."""
+
     tools_mappings = create_tools_mappings(
         anthropic_client=anthropic_client, connection=connection, messages=messages
     )
 
-    response = await words_in_mouth_prompting(
-        anthropic_client=anthropic_client,
+    messages_to_submit = [
+        {"role": item["role"], "content": item["content"]} for item in messages
+    ]
+
+    api_response = await anthropic_client.messages.create(
+        system=system_prompt,
         model=model,
-        system_prompt=system_prompt,
-        appended_user_prompt=APPENDED_USER_PROMPT,
-        start_of_assistant_prompt=START_OF_ASSISTANT_PROMPT,
-        messages=messages,
+        max_tokens=4096,
+        tools=tools,
+        messages=messages_to_submit,
     )
 
-    matches = re.findall(CREATE_TASK_PATTERN, response)
+    messages.append(api_response.to_dict())
 
-    for match in matches:
-        if not isinstance(match, tuple):
-            continue
+    if api_response.stop_reason == "tool_use":
```

Comment:
is this an error condition/situation?

Reply:
Not an error situation, instead this is a case where the model has requested to call a tool that we have defined.